mlflow/pyfunc/__init__.py

```diff
@@ -265,7 +268,51 @@ def _load_model_env(path):
     return _get_flavor_configuration(model_path=path, flavor_name=FLAVOR_NAME).get(ENV, None)
 
 
-def load_model(model_uri, suppress_warnings=True):
+PyFuncOutput = Union[pandas.DataFrame, pandas.Series, np.ndarray, list]
+
+
+class PyFuncModel(object):
+    """
+    MLflow 'python function' model.
+
+    Wrapper around model implementation and metadata loaded with py:func:`mlflow.pyfunc.load_model`.
+    The model implementation can be arbitrary python object returned by the loader_module.
```

Comment:
Wonder if it's worth trying to explain this out a bit more for the end-user (maybe we don't need to, since the class is generally constructed for them / they aren't expected to construct it themselves), e.g. via something like:

```suggestion
    ``model_impl`` can be any Python object that implements the `Pyfunc interface <https://mlflow.org/docs/latest/python_api/mlflow.pyfunc.html#pyfunc-inference-api>`_, and is loaded in `load_model` by invoking the appropriate method within ``loader_module``.
    
    ``model_meta`` is loaded by `:py:func:load_model` from the model metadata (MLmodel) file
```